Modules/Loadable/Markups/VTKWidgets/vtkMRMLMarkupsRegistrationFactory.h

```diff
@@ -0,0 +1,111 @@
+==============================================================================*/
+
+#ifndef __vtkMRMLMarkupsRegistrationFactory_h
+#define __vtkMRMLMarkupsRegistrationFactory_h
+
+#include "vtkSlicerMarkupsModuleMRMLExport.h"
+
+// VTK includes
+#include <vtkObject.h>
+#include <vtkSmartPointer.h>
+
+// STD includes
+#include <map>
+
+class vtkMRMLMarkupsNode;
+class vtkSlicerMarkupsWidget;
+
+/// \brief Class to register Markups so they can be utilized by different
+/// components in the Markups module
+///
+/// This singleton class keeps the registration of markups
+/// Singleton pattern adopted from vtkSegmentationConverterFactory
+class VTK_SLICER_MARKUPS_MODULE_MRML_EXPORT vtkMRMLMarkupsRegistrationFactory : public vtkObject
+{
+public:
+
+  vtkTypeMacro(vtkMRMLMarkupsRegistrationFactory, vtkObject);
+  void PrintSelf(ostream& os, vtkIndent indent) override;
+
+public:
+  /// Return the singleton instance with no reference counting.
+  static vtkMRMLMarkupsRegistrationFactory* GetInstance();
+
+  /// This is a singleton pattern New.  There will only be ONE
+  /// reference to a vtkMRMLMarkupsRegistrationFactory object per process.
+  /// Clients that call this must call Delete on the object so that the reference
+  /// counting will work. The single instance will be unreferenced when
+  /// the program exits.
+  static vtkMRMLMarkupsRegistrationFactory* New();
+
+  /// This reigsters a new markup.
+  /// \param markupsNode vtkMRMLMarkups node to register.
+  /// \param markupsWidget vtkSlicerMarkupsWidget associated to the markup node.
+  void RegisterMarkup(vtkMRMLMarkupsNode *markupsNode,
+                      vtkSlicerMarkupsWidget*markupsWidget);
+
+  /// This returns an isntance to a corresponding vtkSlicerMarkupsWidget associated
+  /// to the MRML node class specified.
+  /// \param nodeClass registered class to retrieve the associated widget.
+  /// \return pointer to associated vtkSLicerMarkupsWidget or nullptr if the MRML node
+  /// class is not registered.
+  vtkSlicerMarkupsWidget* GetWidgetByMarkupsNodeClass(const std::string& nodeClass) const;
+
+  /// This checks if the MRML node specified is registered.
+  /// \param nodeClass MRML node class name to check.
+  /// \return true, if the node is registered in the factory, fasle otherwise.
+  bool IsRegistered(const std::string& nodeClass) const;
+
+protected:
+  vtkMRMLMarkupsRegistrationFactory();
+  ~vtkMRMLMarkupsRegistrationFactory() override;
+  vtkMRMLMarkupsRegistrationFactory(const vtkMRMLMarkupsRegistrationFactory&);
+  void operator=(const vtkMRMLMarkupsRegistrationFactory&);
+
+  // Singleton management functions.
+  static void classInitialize();
+  static void classFinalize();
+
+  friend class vtkMRMLMarkupsRegistrationFactoryInitialize;
+  typedef vtkMRMLMarkupsRegistrationFactory Self;
+
+  // Keeps track of the registered nodes and corresponding widgets
+  std::map<std::string, vtkSmartPointer<vtkSlicerMarkupsWidget>> MarkupsWidgetsMap;
```

Comment:
As I remember if you don't add a space between ">>" then Linux compilation will fail. So instead:

`std::map<std::string, vtkSmartPointer<vtkSlicerMarkupsWidget> > MarkupsWidgetsMap;`

Reply:
I think this is not an issue anymore in recent gcc versions.

Reply:
As @lassoan points out this is not an issue in recent gcc versions. The question is whether it makes sense to have it still as `> >` to make it more backwards-compatible. Any opinions @lassoan @cpinter?

Reply:
I would favor consistency and suggest to use  `> >`

Then, let's create an issue documenting the fact we need to update the entire code base replacing `> >` with `>>`